### PR TITLE
Display pretty alert popups when reaching data sources limits

### PR DESCRIPTION
**Pull request by philipperolet**
Task [here](https://github.com/dust-tt/tasks/issues/133), issue [here](https://github.com/dust-tt/decisions/issues/101)

## Notes
- **Google Drive** Shows the alert after the disclosure, IMHO it is fine
- **Alert limits** Currently different from what is written in our plans; not in this PR's scope, I will follow up with a slack thread
- managed.tsx diff => looks large, but due to a change in identation / div wrapping => fear not :)
image
---
image
---
image

### CODE REVIEW

File: front/pages/w/[wId]/assistant/new.tsx [hunk 1]
```diff
@@ -424,3 +424,25 @@ const AvatarListItem = function ({
...
+}) {
+  const router = useRouter();
+  return (
+    <Popup
+      show={planLimitReached}
+      chipLabel="Free plan"
+      description="Looks like you've used up all your messages. Check out our paid plans to get unlimited messages."
+      buttonLabel="Check Dust plans"
+      buttonClick={async () => {
+        await router.push(`/w/${workspaceId}/subscription`);
```
Comment (fontanierh) [inline front/pages/w/[wId]/assistant/new.tsx:443]: nit: async + await not doing anything here except adding slight overhead

File: front/pages/w/[wId]/builder/data-sources/[name]/index.tsx [hunk 2]
```diff
@@ -240,7 +242,21 @@ function StandardDataSourceView({
...
             <div className="">
-              <div className="mt-0 flex-none">
+              <div className="relative mt-0 flex-none">
+                <Popup
+                  show={showDocumentsLimitPopup}
+                  chipLabel="Free plan"
+                  description="You have reached the limit of documents per data source for the free plan. Upgrade to a paid plan to add more documents."
+                  buttonLabel="Check Dust plans"
+                  buttonClick={async () => {
+                    await router.push(`/w/${owner.sId}/subscription`);
```
Comment (fontanierh) [inline front/pages/w/[wId]/builder/data-sources/[name]/index.tsx:252]: same nit

File: front/pages/w/[wId]/builder/data-sources/static.tsx [hunk 3]
```diff
@@ -113,26 +114,41 @@ export default function DataSourcesView({
...
+              title=""
+              description=""
+              action={
+                !readOnly
+                  ? {
+                      label: "Add a new Data Source",
+                      variant: "primary",
+                      icon: PlusIcon,
+                      onClick: async () => {
+                        await handleCreateDataSource();
```
Comment (fontanierh) [inline front/pages/w/[wId]/builder/data-sources/static.tsx:128]: can we add `useSubmitFunction` to this ? 🙏

**Review by fontanierh — DISMISSED** [review]
LGTM just a nit + a suggestion but no need for review again 🙏

**Comment by philipperolet** [comment]
All  adressed! Thanks

**Comment by spolu** [comment]
Naice

**Pull request merged**